spec/lib/service_spec.rb

```diff
@@ -58,4 +58,38 @@ def generic_step(default_arg: 2)
       end
     end
   end
+
+  describe "Parameters handling" do
+    subject(:result) { service_class.call(**args) }
+
+    context "when calling the service without any params" do
+      let(:args) { {} }
+
+      it "instantiate a default params object" do
+        expect(result[:params]).not_to be_nil
+      end
+    end
+
+    context "when calling the service with params" do
+      let(:args) { { params: { param1: "one" } } }
+
+      context "when there is no `params` step defined" do
+        it "allows accessing `params` through methods" do
+          expect(result[:params].param1).to eq("one")
```

Comment:
wondering if we shouldnt call it raw_params to avoid confusion in this case

Reply:
What do you mean? Having `raw_params` to be able to access the params that were provided? And still have `params` as the ones we processed?
My understanding was that we didn’t have a good use case for wanting to access the raw params directly. And here we still have access to them actually, it’s just we’re accessing them like we do with a contract, so their values aren’t touched at that point. Then, if we want to manipulate them, a contract is probably the best way to do it.

Reply:
It's just not explicit/clear, why one case you do params[:xxx] and the other case params.xxx. You know that's because you didn't define a contract but that's a super subtle implementation detail to know.

If we don't have a good use case for accessing params[:xxx] then just completely disallow it?

Reply:
With the changes brought by the PR, you can’t access `params[:key]` anymore, you have to do `params.key` :slightly_smiling_face:

Reply:
ok ok I see